src/redux/reducers/user.reducer.js

```diff
@@ -0,0 +1,72 @@
+import { getLocal, removeLocal, setLocal } from "../../helper";
+import { FAIL, REQUEST, SUCCESS, USER_ACTION } from "../constants";
+
+const initialState = {
+  token: getLocal("token") || "",
+  userInfo: {},
+  loading: false,
+  fullName: "",
+};
+
+function reducer(state = initialState, action) {
+  switch (action.type) {
+    case REQUEST(USER_ACTION.SIGN_IN):
+      return {
+        ...state,
+        loading: true,
+      };
+    case SUCCESS(USER_ACTION.SIGN_IN):
+      setLocal({ key: "token", value: action?.payload?.data?.accessToken });
+      setLocal({ key: "user", value: { id: action?.payload?.data?.user?.id } });
```

Comment:
```suggestion
      const { accessToken, user } = action.payload.data
      setLocal({ key: "token", value: accessToken });
      setLocal({ key: "user", value: { id: user.id } });
```
đang hơi bị lạm dụng toán tử `?.` đó. Ở đây payload được truyền tới khi đã success, thì có nghĩa là các dữ liệu đây nó sẽ có hết cả, nên k cần phải `?.` làm gì.
Bằng chứng là cái khúc return ở dưới đâu có cần sử dụng tới `?.` đâu